### PR TITLE
Add retry logic in downloading of caffe2 onnx tests, and skip the tests if downloading fails

### PR DESCRIPTION
**Pull request by houseroad**
To address T37014157.

Since we just add retry logic, shall we still throw the assertion? or skip the tests?

### CODE REVIEW

**Review by zrphercule — APPROVED** [review]
LG

File: caffe2/python/onnx/tests/test_utils.py [hunk 1]
```diff
@@ -41,17 +41,24 @@ def _download(self, model):
         for f in ['predict_net.pb', 'init_net.pb', 'value_info.json']:
             url = getURLFromName(model, f)
             dest = os.path.join(model_dir, f)
-            try:
+            retry_num = 3
```
Comment (zrphercule) [inline caffe2/python/onnx/tests/test_utils.py]: Why we dont set this to 2, and...

File: caffe2/python/onnx/tests/test_utils.py [hunk 2]
```diff
@@ -41,17 +41,24 @@ def _download(self, model):
...
+                        downloadFromURLToFile(url, dest,
+                                              show_progress=False)
+                    except TypeError:
+                        # show_progress not supported prior to
+                        # Caffe2 78c014e752a374d905ecfb465d44fa16e02a28f1
+                        # (Sep 17, 2017)
+                        downloadFromURLToFile(url, dest)
+                    break
+                except Exception as e:
+                    if count == retry_num - 1:
```
Comment (zrphercule) [inline caffe2/python/onnx/tests/test_utils.py]: ... set this to count == retry_num -1 = =

File: caffe2/python/onnx/tests/test_utils.py [hunk 3]
```diff
@@ -41,17 +41,24 @@ def _download(self, model):
...
-                    # Caffe2 78c014e752a374d905ecfb465d44fa16e02a28f1
-                    # (Sep 17, 2017)
-                    downloadFromURLToFile(url, dest)
-            except Exception as e:
-                print("Abort: {reason}".format(reason=e))
-                print("Cleaning up...")
-                deleteDirectory(model_dir)
-                raise AssertionError("Test model downloading failed")
+                    try:
+                        downloadFromURLToFile(url, dest,
```
Comment (yinghai) [inline caffe2/python/onnx/tests/test_utils.py]: Why do we put the retry logic inside `downloadFromURLToFile`?

**Review by bddppq — COMMENTED** [review]
I feel it's better to isolate out the retry logic like in onnx https://github.com/onnx/onnx/blob/18f751a/onnx/backend/test/runner/__init__.py#L34-L49

@zrphercule btw. I noticed there is a typo in the onnx function name (excute -> execute)

File: caffe2/python/onnx/tests/test_utils.py (hunk 1 above)
Comment (zrphercule) [reply caffe2/python/onnx/tests/test_utils.py]: also, wonder if twice is enough

**Comment by zrphercule** [comment]
@bddppq uh-oh, a mistake I made when I was young...

**Comment by houseroad** [comment]
@bddppq yeah, onnx's retry logic looks better and more organized. I will follow it. After some thoughts, I think we can still throw exceptions with new retry, and if we still see those tests are flaky, I will skip them.

**Comment by garymm** [comment]
Should we close this or is this flakiness still an issue?

**Pull request closed without merging**